### PR TITLE
Change TS0601_soil_2 to NAS-STH01B2, and fix the dpid issues

### PR DESCRIPTION
**Pull request by baggiowu**
Changed model TS0601_soil_2 to NAS-STH01B2, and fix the TUYA DPID issues.

### CODE REVIEW

**Comment by rare-magma** [comment]
@baggiowu I think it would be better to create another definition for the model you're having issues with. This converter is working fine for my device

File: src/devices/tuya.ts [hunk 1]
```diff
@@ -2043,7 +2042,7 @@ const definitions: DefinitionWithExtend[] = [
                         cancel: tuya.enum(2),
                     }),
                 ],
-                [3, 'soil_moisture', tuya.valueConverter.raw],
+                [3, 'humidity', tuya.valueConverter.raw],
```
Comment (Koenkk) [inline src/devices/tuya.ts:2045]: Since this is a soil moisturising sensor, `soil_moisture ` looks fine to me

**Comment by rare-magma** [comment]
@Koenkk / @baggiowu `humidity`, `battery_state` and `battery` are always reporting the same value on my device so I'm not sure if they actually work. The rest of the datapoints and calibration settings seem to be fine.

File: src/devices/tuya.ts (hunk 1 above)
Comment (baggiowu) [reply src/devices/tuya.ts:2045]: I think so too. But I've received some emails asking to change to humidity.

**Comment by baggiowu** [comment]
> @Koenkk / @baggiowu `humidity`, `battery_state` and `battery` are always reporting the same value on my device so I'm not sure if they actually work. The rest of the datapoints and calibration settings seem to be [fine.](url)

image

**Pull request closed without merging**